File: docs/evaluator/typescript-example/llm.ts
// OpenAI 호환 API 호출 한 개. 앱 코드와 판사가 같이 쓴다.
// 실제 서비스: OPENAI_BASE_URL(예: https://api.openai.com/v1 또는 사내 vLLM), OPENAI_API_KEY 환경변수.
// 외부 패키지 없음 — Node 22+ 내장 fetch. 실행: node --test answers.test.ts (Node >= 22.18 타입 스트리핑)
export type Message = { role: "system" | "user" | "assistant"; content: string };

const BASE_URL = process.env.OPENAI_BASE_URL ?? "http://localhost:8000/v1";
const API_KEY = process.env.OPENAI_API_KEY ?? "none";
const MODEL = process.env.LLM_MODEL ?? "gpt-4o-mini";

export async function chat(messages: Message[], temperature = 0.7, jsonMode = false): Promise<string> {
  const body: Record<string, unknown> = { model: MODEL, messages, temperature };
  if (jsonMode) body.response_format = { type: "json_object" };
  const r = await fetch(`${BASE_URL.replace(/\/$/, "")}/chat/completions`, {
    method: "POST",
    headers: { "Content-Type": "application/json", Authorization: `Bearer ${API_KEY}` },
    body: JSON.stringify(body),
  });
  if (!r.ok) throw new Error(`llm ${r.status}`);
  const data = (await r.json()) as { choices: { message: { content: string } }[] };
  return data.choices[0].message.content;
}

File: docs/evaluator/python-example/llm.py
"""OpenAI 호환 API 호출 한 개. 앱 코드와 판사가 같이 쓴다.
실제 서비스: OPENAI_BASE_URL(예: https://api.openai.com/v1 또는 사내 vLLM), OPENAI_API_KEY 환경변수.
"""
import json, os, urllib.request

BASE_URL = os.environ.get("OPENAI_BASE_URL", "http://localhost:8000/v1")
API_KEY = os.environ.get("OPENAI_API_KEY", "none")
MODEL = os.environ.get("LLM_MODEL", "gpt-4o-mini")


def chat(messages, temperature=0.7, json_mode=False):
    body = {"model": MODEL, "messages": messages, "temperature": temperature}
    if json_mode:
        body["response_format"] = {"type": "json_object"}
    req = urllib.request.Request(
        BASE_URL.rstrip("/") + "/chat/completions",
        data=json.dumps(body).encode(),
        headers={"Content-Type": "application/json", "Authorization": f"Bearer {API_KEY}"},
    )
    with urllib.request.urlopen(req, timeout=60) as r:
        return json.load(r)["choices"][0]["message"]["content"]
